services/worker/src/worker/job_runners/split/duckdb_index.py

```diff
@@ -0,0 +1,280 @@
+    duckdb.execute(LOAD_EXTENSION_COMMAND.format(extension="httpfs"))
+    duckdb.execute(INSTALL_EXTENSION_COMMAND.format(extension="fts"))
+    duckdb.execute(LOAD_EXTENSION_COMMAND.format(extension="fts"))
+
+    # index all columns
+    con = duckdb.connect(str(db_location))
+    con.sql(CREATE_SEQUENCE_COMMAND)
+    con.sql(f"{CREATE_TABLE_COMMAND} read_parquet({parquet_urls});")
+
+    # TODO: by default, 'porter' stemmer is being used, use a specific one by dataset language in the future
+    # see https://duckdb.org/docs/extensions/full_text_search.html for more deails about 'stemmer' parameter
+    con.sql(CREATE_INDEX_COMMAND)
+
+    # create the target revision if it does not exist yet (clone from initial commit to avoid cloning all repo's files)
+    hf_api = HfApi(endpoint=hf_endpoint, token=hf_token)
+    committer_hf_api = HfApi(endpoint=hf_endpoint, token=committer_hf_token)
+    index_file_location = f"{config}/{dataset}-{split}.db"
+    try:
+        refs = hf_api.list_repo_refs(repo_id=dataset, repo_type=DATASET_TYPE)
+        if all(ref.ref != target_revision for ref in refs.converts):
+            initial_commit = hf_api.list_repo_commits(repo_id=dataset, repo_type=DATASET_TYPE)[-1].commit_id
+            committer_hf_api.create_branch(
+                repo_id=dataset, branch=target_revision, repo_type=DATASET_TYPE, revision=initial_commit, exist_ok=True
+            )
+    except RepositoryNotFoundError as err:
+        raise DatasetNotFoundError("The dataset does not exist on the Hub.") from err
+
+    target_dataset_info = hf_api.dataset_info(repo_id=dataset, revision=target_revision, files_metadata=False)
+    all_repo_files: Set[str] = {f.rfilename for f in target_dataset_info.siblings}
+    delete_operations: List[CommitOperation] = []
+    if index_file_location in all_repo_files:
+        delete_operations.append(CommitOperationDelete(path_in_repo=index_file_location))
+
+    # send the files to the target revision
+    add_operations: List[CommitOperation] = [
+        CommitOperationAdd(path_in_repo=index_file_location, path_or_fileobj=db_location)
+    ]
+
+    committer_hf_api.create_commit(
+        repo_id=dataset,
+        repo_type=DATASET_TYPE,
+        revision=target_revision,
+        operations=delete_operations + add_operations,
+        commit_message=commit_message,
+        parent_commit=target_dataset_info.sha,
+    )
+
+    # call the API again to get the list of parquet files
+    target_dataset_info = hf_api.dataset_info(repo_id=dataset, revision=target_revision, files_metadata=True)
+    repo_files = [
+        repo_file for repo_file in target_dataset_info.siblings if repo_file.rfilename == index_file_location
+    ]
+
+    if not repo_files:
+        raise NotAvailableIndexFileError("No index file was found")
+
+    if len(repo_files) != 1:
+        logging.warning(f"Found {len(repo_files)} index files, should be only 1")
+
+    remove_dir(dir_path)
```

Comment:
OK, good!
I think we should reuse the mechanism in https://github.com/huggingface/datasets-server/blob/main/services/worker/src/worker/job_runners/_datasets_based_job_runner.py to avoid concurrency issues.

Reply:
Done in Done in https://github.com/huggingface/datasets-server/pull/1296/commits/3c9b4eef5f394e3f8dc0b60f7c740cfbdfb666b6